packages/promise-kit/index.js

```diff
@@ -0,0 +1,89 @@
+/* global globalThis */
+// @ts-check
+
+// eslint-disable-next-line spaced-comment
+/// <reference types="ses"/>
+
+/** @type {PromiseConstructor} */
+const BestPipelinablePromise = globalThis.HandledPromise || Promise;
+
+/**
+ * @template T
+ * @typedef {Object} PromiseKit A reified Promise
+ * @property {(value: ERef<T>) => void} resolve
+ * @property {(reason: any) => void} reject
```

Comment:
Since TypeScript is purposely unsound, in order to create a better user experience for the typical case, I wonder if we should 
```suggestion
 * @property {(reason: Error) => void} reject
```
to strongly suggest that rejecting with a non-error is probably a mistake.

Reply:
Typescript seem pretty opiniated about typing caught things as `unknown`, so if any changes were made, I'd prefer `unknown` to `any`.